pilot/pkg/xds/sds.go

```diff
@@ -78,14 +84,16 @@ func sdsNeedsPush(updates model.XdsUpdates) bool {
 // Invalid resource names are ignored
 func (s *SecretGen) parseResources(names []string, proxy *model.Proxy) []SecretResource {
 	res := make([]SecretResource, 0, len(names))
+	pkpConf := proxy.Metadata.ProxyConfigOrDefault(s.meshConfig.GetDefaultConfig()).GetPrivateKeyProvider()
```

Comment:
nit: we could just do `proxy.Metadata.ProxyConfig`; if meshConfig changes it invalidates the entire cache.

Reply:
Good idea, fixed :)